### PR TITLE
explicitly adjust lr according to current epoch

### PR DESCRIPTION
**Pull request by hunterkun**

### CODE REVIEW

**Comment by nzmora** [comment]
Nice and simple - I like this solution to issue #70.
I'll do some testing before merging.
Thanks!
Neta

**Comment by netaz** [comment]
@guyjacob What do you think?  This is not perfect, but it's better than the current situation.

**Comment by guyjacob** [comment]
TL;DR - I think we can go with this solution for now, even though it doesn't a specific scenario (details below). The proper solution we should implement is saving and loading the optimizer state.

It seems to me that the only case where this solution will not work is if the LR scheduler updates the LR "progressively". That is, the calculation is based on the current LR and not solely on the initial LR. In that case, if we don't restore the optimizer state as well (which contains the last updated LR), we would get unexpected results.
I checked, and as far as I can see the only scheduler that does such "progressive" updates is [`ReduceLROnPlateau`](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L254) (this includes the schedulers in PyTorch and in Distiller).
So I think we can live with this corner case for now, but as I wrote above, eventually we should implement proper save/load of optimizer state.

**Comment by guyjacob** [comment]
Closing in favor of #74
